conf/gulp-tasks/template/createprecompiletemplates.js

```diff
@@ -28,25 +32,49 @@ function createPrecompileTemplatesTask (locale, translator) {
  *
  * @param {Function} callback called when the task is finished
  * @param {string} locale
+ * @param {boolean} isSearchBarOnly If only the templates related to the SearchBar
+ *                                  should be included.
  * @param {Translator} translator
  * @returns {stream.Readable}
  */
-function _precompileTemplates (callback, locale, translator) {
-  const precompiledFileName = getPrecompiledFileName(locale);
+function _precompileTemplates (callback, locale, isSearchBarOnly, translator) {
+  const precompiledFileName = getPrecompiledFileName(locale, isSearchBarOnly);
   const processAST = ast => new TranslateHelperVisitor(translator).accept(ast);
-  return precompileTemplates(callback, precompiledFileName, processAST);
+
+  return precompileTemplates(
+    callback, isSearchBarOnly, precompiledFileName, processAST);
 }
 
 /**
- * Precopmiles SDK templates to the given output file.
+ * Precopmiles the requested templates to the given output file.
  *
  * @param {Function} callback called when the task is finished
+ * @param {boolean} isSearchBarOnly A boolean indicating if only those templates for
+ *                                  the search-bar integration should be included.
  * @param {string} outputFile
  * @param {Function} processAST a function that takes in and mutates a handlebars AST
  * @returns {stream.Readable}
  */
-function precompileTemplates (callback, outputFile, processAST) {
-  return src('./src/ui/templates/**/*.hbs')
+function precompileTemplates (callback, isSearchBarOnly, outputFile, processAST) {
+  let templatesStream = src('./src/ui/templates/**/*.hbs');
+  let handlebarsWrapperData;
+
+  if (isSearchBarOnly) {
+    templatesStream =
+      templatesStream.pipe(filter(['**/search/**/*.hbs', '**/icons/**/*.hbs']));
+    handlebarsWrapperData = {
+      importStatements: "import Handlebars from 'handlebars/dist/handlebars.runtime.min.js';"
```

Comment:
would it be simpler to pass a boolean into the template and use if statements within the template? I think something like this could work? https://stackoverflow.com/questions/7230470/how-to-use-if-statements-in-underscore-js-templates

Reply:
I attempted to right a JS conditional in the template that would conditionally import one versus the other. Unfortunately, getting conditional ES6 imports to work is difficult and seems like it would entail a bigger refactor of the whole template. I trying using `require`, which works nicely with conditionals, but we're expecting this file to be ES6 apparently.

Reply:
I was thinking we could do something like the below in handlebarswrapper.txt? Unless I'm misunderstanding something and this isn't possible (or we just prefer the injecting the strings directly).
```
<% if (data.isSearchBarOnly) { %>
  import Handlebars from 'handlebars/dist/handlebars.runtime.min.js';
<% } else { %>
  import Handlebars from 'handlebars/dist/handlebars.min.js;
  import templateHelpers from 'template-helpers';
<% } %>
```